### PR TITLE
add partner sharing album

### PR DESCRIPTION
**Pull request by mrwulf**
Having a partner album makes it easy to share it back with your partner

### CODE REVIEW

File: cmdupload/upload.go [hunk 1]
```diff
@@ -334,35 +346,26 @@ func (app *UpCmd) UploadAsset(ctx context.Context, a *assets.LocalAssetFile) {
...
+		if app.PartnerAlbum != "" && a.FromPartner {
+			app.AddToAlbum(resp.ID, app.PartnerAlbum)
+		}
 	} else {
 		app.logger.MessageTerminate(logger.Warning, "already exists on the server")
 	}
 }
 
 func (app *UpCmd) AddToAlbum(ID string, album string) {
-	if !app.CreateAlbums && !app.CreateAlbumAfterFolder && len(app.ImportIntoAlbum) == 0 {
```
Comment (simulot) [inline cmdupload/upload.go:349]: thank you for having cleaned this up.

File: cmdupload/upload.go [hunk 2]
```diff
@@ -263,7 +274,8 @@ func NewUpCmd(ctx context.Context, ic *immich.ImmichClient, logger *logger.Logge
 	cmd.BoolVar(&app.KeepTrashed, "keep-trashed", false, "Import also trashed items")
 	cmd.BoolVar(&app.KeepPartner, "keep-partner", true, "Import also partner's items")
 	cmd.BoolVar(&app.CreateAlbumAfterFolder, "create-album-folder", false, "Create albums for assets based on the parent folder or a given name")
-	cmd.StringVar(&app.ImportIntoAlbum, "album", "", "All assets will be added to this album.")
+	cmd.StringVar(&app.ImportIntoAlbum, "album", "", "All assets will be added to this album. (Must already exist)")
```
Comment (simulot) [inline cmdupload/upload.go:277]:
```suggestion
	Must already exist
```
Not true. Albums are created on the fly id needed

**Review by simulot — COMMENTED** [review]
Why mentioning that albums must be created upfront?
Have you encountered an issue?

File: cmdupload/upload.go (hunk 2 above)
Comment (mrwulf) [reply cmdupload/upload.go:277]: I didn't track down the root cause, but some albums fail to get created. This happened multiple times and I actually ended up changing upload.go to remove the `return` from line 440 ("can't create the album list from server") so it prints the error, but doesn't stop creating additional albums. The error on the server side when an album fails is a constraint violation on table albums_assets_assets so I'm guessing maybe the list of assets to be added to the album needs to be de-duped.

Comment (mrwulf) [reply cmdupload/upload.go:277]: When an album fails to get created the server throws a 500 error, but the message to the client doesn't have any details.

Comment (simulot) [reply cmdupload/upload.go:277]: I'm trying to replicate your situation. How you end up with duplicates in the albums? I suspect that assets are present in several albums. Is that true?

Comment (mrwulf) [reply cmdupload/upload.go]: yes, 84k assets into ~60 albums. Most assets were in 3+ albums

**Pull request merged**